psql2mysql.py

```diff
@@ -347,22 +377,35 @@ def do_migration(config):
 
     logging.setup(cfg.CONF, 'pg2my')
 
-    # restrict the source databsae to postgresql for now
-    if uri_reference(cfg.CONF.source).scheme != "postgresql":
-        print('Error: Only "postgresql" is supported as the source database '
-              'currently', file=sys.stderr)
-        sys.exit(1)
-
-    if (cfg.CONF.target):
-        uri = uri_reference(cfg.CONF.target)
-        if uri.scheme != "mysql+pymysql":
-            print('Error: Only "mysql" with the "pymysql" driver is supported '
-                  'as the target database currently',
-                  file=sys.stderr)
-            sys.exit(1)
-        if (uri.query is None) or ("charset=utf8" not in uri.query.split('&')):
-            print('Error: The target connection is missing the "charset=utf8" '
-                  'parameter.', file=sys.stderr)
-            sys.exit(1)
+    # We expect batch file with this syntax:
+    #
+    # keystone:
+    #   source: postgresql://keystone:p@192.168.243.86/keystone
+    #   target: mysql+pymysql://keystone:p@192.168.243.87/keystone?charset=utf8
+    # cinder:
+    #   source: postgresql://cinder:idRll2gJPodv@192.168.243.86/cinder
+    #   target:
+    if cfg.CONF.batch:
+        try:
+            fh = open(cfg.CONF.batch, 'r')
```

Comment:
You don't close this file after you're done with it.

I would much rather this was redone with a context manager:

```
with open(cfg.CONF.batch, 'r') as f:
    for db_name, db in yaml.load(f).iteritems():
        ...
```